File: checkoutsys-ui/checkoutsys.ui/src/app/models/product.ts
import { OrderProducts } from "./order-products";
import { User } from "./user";

export class Product {
    id: number = 0;
    admindId: number = 0;
    name = "";
    details = "";
    price: number = 0;
    stock: number = 0;
    admin?: User;
    orderProducts: OrderProducts[] = [];
}

File: checkoutsys-api/checkoutsys.api/checkoutsys.api/Models/Product.cs
using System;
using System.Collections.Generic;

namespace checkoutsys.api.Models;

public partial class Product
{
    public long Id { get; set; }

    public long AdminId { get; set; }

    public string Name { get; set; } = null!;

    public string Details { get; set; } = null!;

    public decimal Price { get; set; }

    public long Stock { get; set; }

    public virtual User Admin { get; set; } = null!;

    public virtual ICollection<OrderProducts> OrdersProducts { get; } = new List<OrderProducts>();
}
